clippy_lints/src/time_subtraction.rs

```diff
@@ -132,23 +135,30 @@ fn print_manual_instant_elapsed_sugg(cx: &LateContext<'_>, expr: &Expr<'_>, sugg
     );
 }
 
-fn print_unchecked_duration_subtraction_sugg(
+fn print_unchecked_time_subtraction_sugg(
     cx: &LateContext<'_>,
     left_expr: &Expr<'_>,
     right_expr: &Expr<'_>,
     expr: &Expr<'_>,
 ) {
     let mut applicability = Applicability::MachineApplicable;
+    let ty = cx.typeck_results().expr_ty(expr).peel_refs();
+    let (left_default, left_ty) = match ty::get_type_diagnostic_name(cx, ty) {
+        Some(v) => (format!("<{}>", v.as_str().to_lowercase()), v.as_str().to_string()),
+        None => {
+            return;
+        },
+    };
 
     let ctxt = expr.span.ctxt();
-    let left_expr = snippet_with_context(cx, left_expr.span, ctxt, "<instant>", &mut applicability).0;
+    let left_expr = snippet_with_context(cx, left_expr.span, ctxt, &left_default, &mut applicability).0;
```

Comment:
You should use the `Sugg` interface for getting snippets and call `.maybe_par()` as this expression will be used as a receiver of the `check_sub()` call.

If you don't, the expression `2 * duration - duration` will be fixed into `2 * duration.checked_sub(duration).unwrap()` which is incorrect.